File: api/balance.js
export default async function handler(req, res) {
    const { symbol } = req.query;
    if (!symbol) return res.status(400).json({ error: 'Missing symbol' });
    const sym = symbol.toUpperCase();

    if (process.env.FINANCIAL_API_KEY) {
        try {
            const url = `https://api.financialdata.net/api/v1/balance-sheet/${sym}?limit=1&token=${process.env.FINANCIAL_API_KEY}`;
            const data = await fetch(url).then(r => r.json());
            if (data?.financials?.length) {
                const bs = data.financials[0];
                return res.json({
                    totalAssets: bs.totalAssets,
                    totalDebt: bs.totalDebt,
                    cashAndCashEquivalents: bs.cashAndCashEquivalents,
                    totalEquity: bs.totalEquity,
                });
            }
        } catch (e) { console.warn('FinancialData balance failed', e.message); }
    }

    if (process.env.FMP_API_KEY) {
        try {
            const url = `https://financialmodelingprep.com/api/v3/balance-sheet-statement/${sym}?limit=1&apikey=${process.env.FMP_API_KEY}`;
            const data = await fetch(url).then(r => r.json());
            if (data?.length) {
                const bs = data[0];
                return res.json({
                    totalAssets: bs.totalAssets,
                    totalDebt: bs.totalDebt,
                    cashAndCashEquivalents: bs.cashAndCashEquivalents,
                    totalEquity: bs.totalEquity,
                });
            }
        } catch (e) { console.warn('FMP balance failed', e.message); }
    }

    return res.status(503).json({ error: 'No balance sheet data available' });
}

File: api/balance.py
import json
from urllib.request import Request, urlopen
from urllib.parse import urlencode

FINANCIAL_API_KEY = "afb05902415e550cb4a4cccefa62d2a1"
FINANCIAL_BASE = "https://api.financialdata.net/v3"

def handler(request):
    query = request.get_query() or {}
    symbol = query.get("symbol", [None])[0]
    if not symbol:
        return {"statusCode": 400, "body": json.dumps({"error": "Missing symbol"})}
    
    params = urlencode({"limit": 1, "token": FINANCIAL_API_KEY})
    url = f"{FINANCIAL_BASE}/financials/balance-sheet/{symbol.upper()}?{params}"
    req = Request(url)
    try:
        with urlopen(req, timeout=10) as resp:
            data = json.loads(resp.read().decode())
        if "financials" in data and data["financials"]:
            bs = data["financials"][0]
            result = {
                "totalAssets": bs.get("totalAssets"),
                "totalDebt": bs.get("totalDebt"),
                "cashAndCashEquivalents": bs.get("cashAndCashEquivalents"),
                "totalEquity": bs.get("totalEquity"),
            }
            return {"statusCode": 200, "body": json.dumps(result), "headers": {"Content-Type": "application/json"}}
        else:
            return {"statusCode": 404, "body": json.dumps({"error": "No balance sheet"})}
    except Exception as e:
        return {"statusCode": 500, "body": json.dumps({"error": str(e)})}
